### PR TITLE
SPLICE-1905: Failed to run select query from partitioned external table in case of NULL presence.(2.6)

### PR DESCRIPTION
**Pull request by stpabhi**

### CODE REVIEW

**Comment by stpabhi** [comment]
@jyuanca Could you please review and approve?

File: hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java [hunk 1]
```diff
@@ -1928,6 +2020,17 @@ public void testPartitionBySmallIntOrcTable() throws Exception {
         resultString = TestUtils.FormattedResult.ResultFactory.toStringUnsorted(rs);
         assertEquals(expected, resultString);
         rs.close();
+
+        methodWatcher.executeUpdate("insert into t_partition_by_smallint values(NULL, 1, 'DDD')");
+        rs = methodWatcher.executeQuery("select a1, count(*) from t_partition_by_smallint group by a1 order by 1");
+
+        expected = "A1 |B1 |C1  |\n" +
+                "----------------\n" +
+                "   NULL    | 1 |";
```
Comment (yxia92) [inline hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java]: There are 3 rows inserted earlier before the row with null on the partition column is inserted, so the expected result is wrong. It should be:
        expected = "A1  | 2 |\n" +
                "----------\n" +
                "  1  | 2 |\n" +
                "  2  | 1 |\n" +
                "NULL | 1 |";

Comment (stpabhi) [reply hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java]: Thanks. Will fix it.

Comment (yxia92) [reply hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java]: Thanks. Change is only needed for branch-2.6. We have fixed the IT in the master branch.

Comment (stpabhi) [reply hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java]: @yxia92 Fixed

**Comment by stpabhi** [comment]
@OlegMazurov Fixed in 2.6. Test passed.

**Pull request merged**